cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java

```diff
@@ -128,18 +132,45 @@
   }
 
   @Override
+  protected RuntimeInfo createRuntimeInfo(ProgramController controller, Program program) {
+    RunId twillRunId = ((AbstractTwillProgramController) controller).getTwillRunId();
```

Comment:
Better check `instanceOf` because casting. You should see a yellow warning in the IntelliJ for this statement. We should go for all green, instead of full of yellow warning.